scripts/build-container-image.sh

```diff
@@ -47,22 +65,17 @@ git_get_current_commit_short_hash() {
   git rev-parse --short HEAD
 }
 
-git_get_current_branch() {
-  git rev-parse --abbrev-ref HEAD
-}
-
 get_date() {
   # Should this be current date, or latest commit date?
   date +%Y%m%d
 }
 
-get_image_tag() {
-  echo "$(git_get_current_branch)_$(get_date)_$(git_get_current_commit_short_hash)"
-}
-
 cleanup_old_tags() {
-  all_tags=$(list_tags)
-  branch=$(git_get_current_branch)
+  if [[ "${GIT_REFERENCE_TYPE}" != "branch" ]]; then
+    return
+  fi
+  all_tags=($(list_tags))
```

Comment:
```suggestion
  mapfile -t all_tags < <(list_tags)
```
Assigning the command output to an array is problematic, shellcheck recommends `mapfile` which works here nicely.

https://www.shellcheck.net/wiki/SC2207